From 8e9536ec9eefc3c63179d8738160ad361102beef Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 24 Jun 2014 16:10:27 +0200
Subject: [PATCH 3/3] pulse: fix use after free in stream_drain_cb()

---
 src/pulse.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/pulse.c b/src/pulse.c
index 22eab14..27148d4 100644
--- a/src/pulse.c
+++ b/src/pulse.c
@@ -650,13 +650,14 @@ static void stream_drain_cb(pa_stream *s, int success, void *userdata) {
         pa_stream_disconnect(s);
         out->error = err;
         out->finished = TRUE;
-    }
 
-    if (out->drain_operation) {
-        pa_operation_unref(out->drain_operation);
-        out->drain_operation = NULL;
+        if (out->drain_operation) {
+            pa_operation_unref(out->drain_operation);
+            out->drain_operation = NULL;
+        }
     }
 
+
     pa_threaded_mainloop_signal(p->mainloop, FALSE);
 }
 
-- 
1.9.3

