diff -u libcanberra-0.22/Makefile.am libcanberra-0.22.new/Makefile.am
--- libcanberra-0.22/Makefile.am	2009-09-11 23:59:30.000000000 +0200
+++ libcanberra-0.22.new/Makefile.am	2014-06-24 16:25:49.270344615 +0200
@@ -46,7 +46,7 @@
 
 if GCONF_SCHEMAS_INSTALL
 install-data-local:
-                if test -z "$(DESTDIR)" ; then \
+		if test -z "$(DESTDIR)" ; then \
                         for p in $(schemas_DATA) ; do \
                                 GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/data/$$p ; \
                         done \
diff -u libcanberra-0.22/Makefile.in libcanberra-0.22.new/Makefile.in
--- libcanberra-0.22/Makefile.in	2009-10-20 02:29:59.000000000 +0200
+++ libcanberra-0.22.new/Makefile.in	2014-06-24 16:27:45.804399121 +0200
@@ -933,7 +933,7 @@
 
 
 @GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@install-data-local:
-@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                if test -z "$(DESTDIR)" ; then \
+@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@		if test -z "$(DESTDIR)" ; then \
 @GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                        for p in $(schemas_DATA) ; do \
 @GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                                GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/data/$$p ; \
 @GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                        done \
